### DISCUSSION TITLE
Custom "Edit this page" repository links

### DISCUSSION THREAD

**Discussion by sccmcca** [mkdocs/mkdocs#2683] [Q&A]
Is there a configuration or plugin that would allow to define custom links to different source repositories, per page?

In my case I have a bash script that fetches markdown files from multiple repositories and mkdocs renders them as pages. 

If a user wants to edit, I'd like them to be able to click the "Edit this page" button and be directed to the original file on the original repository (different from my site repository). I have no problem defining these manually as I have a minimal number of pages.

Any hints on this? Thanks!

**Comment by oprypin** [comment]
mkdocs-gen-files plugin allows you to copy files over in a way that doesn't make your current checkout "dirty"; the `mkdocs build` is kept self-contained.

That, and it indeed actually allows you to set the edit URI.

https://oprypin.github.io/mkdocs-gen-files/api.html#mkdocs_gen_files.editor.FilesEditor.set_edit_path

**Reply by sccmcca** [reply]
Looks interesting! 

Could you give me a rough data example of what this would look like configured in mkdocs.yml if I wanted to assign an edit path to a file on a GitHub repo, on a specific page in my docs directory?

**Reply by oprypin** [reply]
Only if you provide an example what you have now.
Sorry, I don't think I'll spend time coming up with a example to then base my example on.

For now I can only suggest looking through usages out there: https://github.com/search?q=mkdocs_gen_files&type=code ; https://github.com/search?q=mkdocs_gen_files+set_edit_path&type=code

The mkdocs.yml always looks almost exactly the same, then it's all about what you put into the Python script.
https://github.com/oprypin/mkdocs-gen-files#usage

**Reply by sccmcca** [reply]
I'll take a look. Thanks!
